### ISSUE TITLE
Enable the AdvancedLowering3 test

### ISSUE THREAD

**Issue by naoyam** [csarofeen/pytorch#624]
Root mapping seems to have an issue.

See https://github.com/csarofeen/pytorch/pull/597#discussion_r560349607

**Comment by naoyam** [comment]
The tensors of `AdvancedLowering3` are:

```
%kernel_math {
T2[ sbS5{1}, iS4{i3} ]
   = T0[ sbS0{1}, iS1{i3} ]
   + double(2);
T4[ sbS9{1}, iS8{i3} ]
   = T2[ sbS5{1}, iS4{i3} ]
   + double(4);
T3[ iS6{i5}, iS7{i7} ]
   = T1[ iS2{i5}, iS3{i7} ]
   + double(3);
T5[ iS11{i5}, iS10{i3} ]
   = T2[ sbS5{1}, iS4{i3} ]
   + T3[ iS6{i5}, iS7{i7} ];
}
```

We have a root-mapping failure with:

```
tv0->computeAt(tv4, -1);
```
```
C++ exception with description "false INTERNAL ASSERT FAILED at "../torch/csrc/jit/codegen/cuda/root_domain_map.cpp":446, please report a bug to PyTorch. Mapping IterDomain sbS9{1} of [ sbS9{1}, iS8{i3} ] not possible as it would require recomputing the source tensor. Producer root: sbS5{1} iS4{i3}. Consumer root: sbS9{1} iS8{i3}
```

ComputeAtRootMapping conservatively determines that the first axis of `T2` may not be computed at `T4`. That is because that axis has two distinctive mapping relationships: one with the first axis of `T4` and another with the first axis of `T5`. Since there is no concrete axis mapped with the first `T4`, it assumes it could be mapped with an axis of any size, which prevents mapping with the first `T5` axis. Since the first axis of `T2` has two possible mappable sets, and they can't be mapped together, it decides that sharing that loop is not legally possible. 

However, since the axis of `T4` is still broadcast and never gets concretized, there is no actual loop instantiated, so the concern of "sharing" is actually meaningless. Such a case should be handled as a safe mapping.

**Comment by naoyam** [comment]
Closed with #675
